otel/conf.go

```diff
@@ -0,0 +1,584 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Package otel provides data structures to represent and generate otel configuration.
+package otel
+
+import (
+	"fmt"
+	//	"net"
+	"strings"
+	"text/template"
+)
+
+const (
```

Comment:
Question: Would it be possible to use OTel's YAML structs and then just serialize those directly, instead of defining these text templates?

Reply:
I can give it a try. For the default configuration, e.x. metrics agent itself's metrics which is constant string blob now, How it can be fitted inside the yaml structs. I will let you know how it goes

Reply:
Thinking this can be quite an important discussion; thus i opened a section in the Design Doc discussing it. Also cite the same discussion below as i don't want to link the internal doc bookmark in public.

```
First, OTEL configuration is yaml formatted. It's definitely great that if we can use yaml struct for the implementation which just is more natural way of implementing it. 

However, the[ default 500+ lines windows configuration](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/blob/master/config/config-windows.yaml) just makes it quite cumbersome to convert each single OTEL plugin(receiver, processor, exporter and extension) as a struct. 

To give an example, the complicated way that we have for processing the system metrics in the default config: https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/blob/master/config/config-windows.yaml#L92. Converting this 200+ lines of metricstransformer into a yaml struct will need quite some work (Create each metric its own sub struct type variable then append all of it together under the parent struct variable). Since processes metrics are default and free; also we are not exposing it to the end-user for re-configuring, so do we want to spend time on converting it towards a yaml struct that might/might not be worth it.  

As it's impossible to have the yaml partially made by string blob and paritally made by yaml struct, we need to decide whether we want to use string blob or yaml structs. Thinking about the long term (when customer is able to customize more things), it might be better to spend sometime and implement the yaml struct out despite the work amount of doing so plus it's unseen by client..

I will leave the decision to Ling and Dave.
```

Reply:
Also, we can explore in a follow-up PR, don't need to block this one.

Reply:
SGTM.  left a todo in b/176757501#comment3